lib/middleware/vhost.js

```diff
@@ -15,7 +17,7 @@ module.exports = () =>
       if (bucket) {
         ctx.path = `/${bucket}${ctx.path}`;
       }
-    } else if (ctx.hostname !== "localhost" && ctx.hostname !== "127.0.0.1") {
+    } else if (ctx.hostname !== "localhost" && !net.isIP(ctx.hostname)) {
       // otherwise attempt to distinguish virtual host-style requests
       ctx.path = `/${bucket}${ctx.path}`;
```

Comment:
Thanks for this! Though this doesn't totally fix the underlying issue. I believe this line should also read

```javascript
  ctx.path = `/${ctx.hostname}${ctx.path}`
```

Regarding tests, I'm realizing the existing test should be named <code>should reach the server with a bucket ~~vhost~~ subdomain</code>. The behavior fixed in this PR should probably add a couple tests for vhost-style and IP-style requests.